File: languages/python/01-foundations/formatted-output-and-iomanip/example/main.py
# This example shows formatting values so output is easier to read and compare.
# In Python, the example favors direct readable steps while keeping validation visible.

# Run one direct scenario at the top level so the printed result is easy to verify.
items = [
    ("Notebook", 2, 3.5),
    ("Pencil", 5, 0.8),
    ("Backpack", 1, 29.99),
]

# Print the observed state here so learners can match the code path to the result.
print(f"{'Item':<12}{'Qty':>6}{'Unit':>10}{'Total':>10}")
print("-" * 38)

grand_total = 0.0
for name, qty, unit_price in items:
    total = qty * unit_price
    grand_total += total
    print(f"{name:<12}{qty:>6}{unit_price:>10.2f}{total:>10.2f}")

print("-" * 38)
print(f"{'Grand total':<28}{grand_total:>10.2f}")

File: languages/go/01-foundations/formatted-output-and-iomanip/example/main.go
// This example shows formatting values so output is easier to read and compare.
// In Go, the example keeps the flow explicit through small functions, interfaces, and concrete data.

package main

import "fmt"

// Define the reusable pieces first so the main flow can focus on one observable scenario.
type Item struct {
	Name      string
	Quantity  int
	UnitPrice float64
}

// Run one deterministic scenario so the console output makes formatting values so output is easier to read and compare easy to verify.
func main() {
	// Build the sample state first, then let the later output confirm the behavior step by step.
	items := []Item{
		{"Notebook", 2, 3.5},
		{"Pencil", 5, 0.8},
		{"Backpack", 1, 29.99},
	}

	// Print the observed state here so learners can connect the code path to a concrete result.
	fmt.Printf("%-12s%6s%10s%10s\n", "Item", "Qty", "Unit", "Total")
	fmt.Println("--------------------------------------")

	grandTotal := 0.0
	for _, item := range items {
		total := float64(item.Quantity) * item.UnitPrice
		grandTotal += total
		fmt.Printf("%-12s%6d%10.2f%10.2f\n", item.Name, item.Quantity, item.UnitPrice, total)
	}

	fmt.Println("--------------------------------------")
	fmt.Printf("%-28s%10.2f\n", "Grand total", grandTotal)
}
